### PR TITLE
update aws cli to version 2

### PR DESCRIPTION
**Pull request by pdeveltere**
updated aws cli to v2 and updated some of the deprecated github actions

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,12 +1,8 @@
-FROM python:3.10.0a7-alpine3.13
+FROM alpine:3.19
```
Comment (pdeveltere) [inline Dockerfile:1]: Alpine 3.18 includes aws-cli v2 as a 'main' package, so all you need to do is just run apk add aws-cli

File: entrypoint.sh [hunk 2]
```diff
@@ -18,7 +18,7 @@ if [ -n "${INPUT_AWS_SECRET_ACCESS_KEY:-}" ]; then
 fi
 
 if [ -n "${INPUT_AWS_REGION:-}" ]; then
-  export AWS_DEFAULT_REGION="${INPUT_AWS_REGION}"
+  export AWS_REGION="${INPUT_AWS_REGION}"
```
Comment (pdeveltere) [inline entrypoint.sh:21]: AWS_REGION takes precendent in AWS CLI v2

**Review by ianbelcher — APPROVED** [review]

**Comment by ianbelcher** [comment]
Thanks @pdeveltere this looks like a great upkeep commit, I appreciate it.

**Comment by pdeveltere** [comment]
@ianbelcher perhaps it makes sense to create a new release v2 with this change?

**Pull request merged**